### PR TITLE
Add basic authenticator and file based userstore

### PR DESCRIPTION
**Pull request by isurulucky**

### CODE REVIEW

File: stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authentication/basic/authn-interceptor.bal [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package ballerina.security.authentication.basic;
```
Comment (Shan1024) [inline stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authentication/basic/authn-interceptor.bal]: We need to add license headers to all .bal files.

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/security/AuthorizationTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        registry.initRegistry(getRuntimeProperties());
+        registry.loadConfigurations();
+    }
+
+    private Map<String, String> getRuntimeProperties() {
+        String ballerinaConf = "ballerina.conf";
+        Map<String, String> runtimeConfigs = new HashMap<>();
+        runtimeConfigs.put(ballerinaConf, getClass().getClassLoader()
+                .getResource("datafiles/config/security/authorization/permissionstore/" +
+                        ballerinaConf).getPath());
```
Comment (Shan1024) [inline tests/ballerina-test/src/test/java/org/ballerinalang/test/security/AuthorizationTest.java]: I think we can merge this line with the above line.

File: stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authentication/basic/authn-interceptor.bal [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+BasicAuthenticator authenticator;
+
+@Description {value:"Intercepts a request for authentication"}
+@Param {value:"req: InRequest object"}
+@Return {value:"boolean: true if authentication is a success, else false"}
+public function handle (http:InRequest req) (boolean) {
+    // extract the header value
+    string basicAuthHeaderValue;
+    error err;
+    basicAuthHeaderValue, err = utils:extractBasicAuthHeaderValue(req);
```
Comment (Shan1024) [inline stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authentication/basic/authn-interceptor.bal]: Cant we use `var basicAuthHeaderValue, err = utils:extractBasicAuthHeaderValue(req);` here?

File: stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authentication/basic/authn-interceptor.bal [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            log:printDebug("Auth cache hit for user: " + authInfo.username);
+            return authInfo.isAuthenticated;
+        }
+        // if a casting error occurs, clear the cache entry
+        authenticator.clearCachedAuthResult(basicAuthCacheKey);
+    }
+
+    string username;
+    string password;
+    username, password, err = utils:extractBasicAuthCredentials(basicAuthHeaderValue);
```
Comment (Shan1024) [inline stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authentication/basic/authn-interceptor.bal]: Cant we use `var username, password, err = utils:extractBasicAuthCredentials(basicAuthHeaderValue);` here?

File: stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authorization/authz-checker.bal [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    }
+}
+
+@Description {value:"Error handler"}
+@Param {value:"message: error message"}
+@Return {value:"error: error populated with the message"}
+function handleError (string message) (error) {
+    error e = {message:message};
+    return e;
+}
```
Comment (Shan1024) [inline stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authorization/authz-checker.bal]: Missing new lines at the end of some files.

**Comment by isurulucky** [comment]
Closing the PR to do the changes suggested at a review meeting.

**Pull request closed without merging**